lib/sprite_animation.dart

```diff
@@ -4,6 +4,40 @@ import 'dart:ui';
 import 'extensions/vector2.dart';
 import 'sprite.dart';
 
+class SpriteAnimationOpts {
+  int amount;
+  int amountPerRow;
+  Vector2 texturePosition;
+  Vector2 textureSize;
+  List<double> stepTimes;
```

Comment:
I think this should be a List<Frame> which has all this info. that is how we store it in a unified format.
then we can have a myriad of constructors for this class. for example, sequenced takes amount and a start position and follow as rows. we can rename sequenced to row and have another one for following a column. we can have one that figures everything out, and one that takes the List<Frame> as is.

Reply:
I didn't understood

Reply:
yes. I mean we can have a unified data type to represent the frames. it can contain frames of any size, any position, with any stepTime. So it will be a List<FrameData> where FrameData := { pos, size, stepSize }. Once we have that we can have a plethora of named constructors, like sequenced that takes just one pos, one size + amount and generates the frame data.

That will allow us to add a lot of constructors. I am proposing renaming sequenced to row and create a col one, for example. We can also have other ones.

Reply:
Alright, I think I understood, to be honest, I don't think that it will be better, from a user view. But I will try thinking on something and make some tests and we can discuss later

Reply:
@luanpotter I have made the changes, it is way better now! Thanks for the suggestions, please let me know of what you think of that approach.